docs/userstories/v1/egress-control-in-anp/egress-control-cluster-nodes.md

```diff
@@ -0,0 +1,39 @@
+
+### Summary
+
+1. **As a** cluster administrator **I want** to easily block access from
+cluster workloads to k8s-apiservers running on cluster nodes (host-net pods).
+2. **As a** cluster administrator **I want** to easily block access from
+cluster workloads to specific ports on cluster nodes.
+
+### Description
+
+Currently the behaviour for policies defined around traffic from cluster
+workloads (non-hostNetworked pods) towards hostNetworked pods or nodes in the
+cluster is undefined. See https://github.com/kubernetes-sigs/network-policy-api/issues/73.
+
+ANP currently supports only east-west traffic and this traffic flow cuts from
+overlay to underlay which makes this part of the egress (south-north) use case.
+
+Let's provide a defined behaviour in ANP to explicitly achieve the use cases.
+
+### Acceptance Criteria
+
+1. If I apply a policy that limits egress access from workloads identified
+by label:A towards k8s-apiserver backends (master nodes at 6443 port), it should
+be able to guarantee that those workloads cannot access the apiserver
+endpoints:targetPort. Even if the master node IPs change, the policy should be
+able to automatically pivot towards the new nodeIPs.
+
+[
+Also, here are a few points that need to be addressed:
+
+1. Clarify that the existing `podSelector` in the ANP API are for regular
+non-hostNetworked pods only. These don't apply to hostNetworked pods.
+2. Add a new well defined optional field called `hostNetworkPodSelector` to both
```

Comment:
policy rules will be very tricky to enforce though, since hostnet pods share the same node ip...

Reply:
hmm I agree with you but on L3/L4 level I am hoping because of how we don't have implicit deny's in ANP users will use this along with the port field to say what they want to allow like 6443 for k8s-server?
but it is certainly something to look out for! if accidentally people block everything on this IP, then that does impact the to/from traffic of nodes...

Reply:
> Just to get this down in writing somewhere: it is _theoretically_ possible to distinguish "traffic from hostNetwork pod A", "traffic from hostNetwork pod B", and "traffic from a node process" because each hostNetwork pod has its own cgroup and there are APIs (eg `iptables -m cgroup`) that can match based on the cgroup of the socket associated with a connection.
> 
> AFAIK no one has tried to do this in practice, and the `iptables-extensions` man page suggests that it only works on outbound packets, not inbound ones (though the `nft` man page does not have the same warning and actually gives an example using `input`, so maybe the iptables warning is out of date?)

thanks @danwinship! I am certainly learning a lot here! So seems like this is "tricky" based on implementations but not impossible to implement in the end...

From API perspective - these are egress stories so normal pods (to)-> host-networked pods OR normal pods (to)-> external entities outside the cluster
However I mentioned adding support for host-networked pods (to)-> normal-pods which makes this ingress use case depending on how we look at it but in the end host-networked pods are _in_ the cluster.. Just wondering if there are asks or stories to add support for host-networked pods in the subject field.